### ISSUE TITLE
v1.1.5: InvalidIdeException: IDE by path '/home/plipski/.pluginVerifier/ides' is invalid: IDE directory content is invalid

### ISSUE THREAD

**Issue by PawelLipski** [JetBrains/gradle-intellij-plugin#779]
When running `runPluginVerifier` task.

Only happens after bumping the version to 1.1.5
1.1.4 is not affected yet.

Happens both on my local machine (Fedora) and in CI: https://app.circleci.com/pipelines/github/VirtusLab/git-machete-intellij-plugin/4203/workflows/a46676bc-82bc-4226-a8cf-560640534278/jobs/4422

Logs from my machine:
```
> Task :runPluginVerifier
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by retrofit2.Platform (file:/home/plipski/.gradle/caches/modules-2/files-2.1/org.jetbrains.intellij.plugins/verifier-cli/1.266/8e58667f52e52f4ff6f6126b316409584c3f44bb/verifier-cli-1.266-all.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class,int)
WARNING: Please consider reporting this to the maintainers of retrofit2.Platform
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Exception in thread "main" com.jetbrains.plugin.structure.ide.InvalidIdeException: IDE by path '/home/plipski/.pluginVerifier/ides' is invalid: IDE directory content is invalid
        at com.jetbrains.plugin.structure.ide.IdeManagerImpl.createIde(IdeManagerImpl.kt:34)
        at com.jetbrains.plugin.structure.ide.IdeManagerImpl.createIde(IdeManagerImpl.kt:25)
        at com.jetbrains.pluginverifier.ide.IdeDescriptor$Companion.create(IdeDescriptor.kt:59)
        at com.jetbrains.pluginverifier.options.OptionsParser.createIdeDescriptor(OptionsParser.kt:78)
        at com.jetbrains.pluginverifier.options.OptionsParser.createIdeDescriptor(OptionsParser.kt:73)
        at com.jetbrains.pluginverifier.tasks.checkPlugin.CheckPluginParamsBuilder.build(CheckPluginParamsBuilder.kt:38)
        at com.jetbrains.pluginverifier.tasks.checkPlugin.CheckPluginParamsBuilder.build(CheckPluginParamsBuilder.kt:23)
        at com.jetbrains.pluginverifier.PluginVerifierMain.main(PluginVerifierMain.kt:125)
[gradle-intellij-plugin :git-machete-intellij-plugin:runPluginVerifier] Error during Plugin Verifier CLI execution:
Starting the IntelliJ Plugin Verifier 1.266
Verification reports directory: /home/plipski/git-machete-intellij-plugin/build/reports/pluginVerifier
2021-09-04T19:14:58 [main] INFO  verification - Reading IDE /home/plipski/.pluginVerifier/ides/IC-2020.1.4
2021-09-04T19:14:58 [main] INFO  c.j.p.options.OptionsParser - Reading IDE from /home/plipski/.pluginVerifier/ides/IC-2020.1.4
2021-09-04T19:15:05 [main] INFO  verification - Reading IDE /home/plipski/.pluginVerifier/ides/IC-2020.2.4
2021-09-04T19:15:05 [main] INFO  c.j.p.options.OptionsParser - Reading IDE from /home/plipski/.pluginVerifier/ides/IC-2020.2.4
2021-09-04T19:15:10 [main] INFO  verification - Reading IDE /home/plipski/.pluginVerifier/ides/IC-2020.3.4
2021-09-04T19:15:10 [main] INFO  c.j.p.options.OptionsParser - Reading IDE from /home/plipski/.pluginVerifier/ides/IC-2020.3.4
2021-09-04T19:15:14 [main] INFO  verification - Reading IDE /home/plipski/.pluginVerifier/ides/IC-2021.1.3
2021-09-04T19:15:14 [main] INFO  c.j.p.options.OptionsParser - Reading IDE from /home/plipski/.pluginVerifier/ides/IC-2021.1.3
2021-09-04T19:15:18 [main] INFO  verification - Reading IDE /home/plipski/.pluginVerifier/ides
2021-09-04T19:15:18 [main] INFO  c.j.p.options.OptionsParser - Reading IDE from /home/plipski/.pluginVerifier/ides
```

**Comment by izhangzhihao** [comment]
same issue here: https://github.com/izhangzhihao/intellij-rainbow-brackets/pull/2358/checks?check_run_id=3515939310

**Comment by hsz** [comment]
Thanks for reporting. The issue is fixed with fa1f669639bc81e0365a5741871788a6077183ce
It was caused because of the invalid path returned from the function that downloads IDEs – it was returning parent directory instead of the actual IDE's dir: https://github.com/JetBrains/gradle-intellij-plugin/commit/f0e747161a3b24be9978c99e7f013a22df6cf2e3#diff-d7d397447dff9b82440110ee94dca3dda67b3f9449af7004a72a5f6ea99646beR506-R507

What is interesting here is that this issue was already present since 1.1.4 – probably all the downloaded IDEs were still present and got removed after upgrading to 1.1.5

Anyway, 1.1.6 is out now.
